DoSomething/quasar
# Updating DBT to v0.16, and fixing python-dateutil dependency which is from an old legacy issue.

#### What's this PR do?
* Updates `dbt` with pinning to `0.16` branch (`~=0.16`). 
* Fixes a dependency issue with `python-dateutil` version that we'd locked to `2.6.1` due to a release bug 3 years ago. Also this was mostly used by our legacy data pipeline code. 
#### Where should the reviewer start?
https://github.com/DoSomething/quasar/compare/dbt-update-16?expand=1#diff-1e61a31bf9b94805f869dc4137ec1885
#### How should this be manually tested?
I did an in-place upgrade and fresh install on my local, but feel free to do a `pipenv install` on your local machine. Let's *also* make sure that when deployed to Quasar QA, the local environment picks up the updated changes. We should also run a few DBT jobs to make sure the updated version doesn't break anyhting in our existing models!

#### What are the relevant tickets?
https://www.pivotaltracker.com/story/show/170059406
